lib/nats.js

```diff
@@ -173,51 +175,66 @@ Client.prototype.parseOptions = function(opts) {
       options.url = DEFAULT_PRE + opts.port;
     }
     // Pull out various options here
-    this.assignOption(opts, 'url');
-    this.assignOption(opts, 'uri', 'url');
-    this.assignOption(opts, 'user');
```

Comment:
I've removed user and pass/password because they could lead to incompatible information, such as:
```
url: ['nats://userA:passA@localhost:4222'], user:'userB', pass:'passB'
```
But we could think about other policies (i.e. user and pass in the uri override the options)